modules/005-external-module-manager/hooks/migration_create_update_policies.go

```diff
@@ -0,0 +1,132 @@
+// Copyright 2023 Flant JSC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// This migration hook creates module update policies for all but deckhouse module sources on first run.
+// After creation, d8-system namespace gets annotated and migration doesn't fire up anymore.
+package hooks
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/flant/addon-operator/pkg/module_manager/go_hook"
+	"github.com/flant/addon-operator/sdk"
+	"github.com/flant/shell-operator/pkg/kube/object_patch"
+	"github.com/iancoleman/strcase"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+
+	"github.com/deckhouse/deckhouse/deckhouse-controller/pkg/apis/deckhouse.io/v1alpha1"
+	"github.com/deckhouse/deckhouse/go_lib/dependency"
+)
+
+const (
+	defaultUpdateMode   = "Auto"
+	migrationAnnotation = "modules.deckhouse.io/ensured-update-policies"
+)
+
+var _ = sdk.RegisterFunc(&go_hook.HookConfig{
+	// ensure crds hook has order 5, for creating a module source we should use greater number
+	OnBeforeHelm: &go_hook.OrderedConfig{Order: 6},
```

Comment:
I guess we can use OnStartup here, also actual crd are created on deckhouse-controller startup. Here is only legacy crds, you can ignore them

Reply:
done